templates/repo/settings/options.tmpl

```diff
@@ -80,9 +80,26 @@
 					</div>
 					<div class="field {{if .Err_MirrorAddress}}error{{end}}">
 						<label for="mirror_address">{{.i18n.Tr "repo.mirror_address"}}</label>
-						<input id="mirror_address" name="mirror_address" value="{{.Mirror.FullAddress}}" required>
+						<input id="mirror_address" name="mirror_address" value="{{.Mirror.AddressNoCredentials}}" required>
 						<p class="help">{{.i18n.Tr "repo.mirror_address_desc"}}</p>
 					</div>
+					<div class="ui accordion optional field">
+						<label class="ui title {{if .Err_Auth}}text red active{{end}}">
+							<i class="icon dropdown"></i>
+							<label for="">{{.i18n.Tr "repo.need_auth"}}
+						</label>
+						<div class="content {{if .Err_Auth}}active{{else if .Mirror.Username}}active{{end}}">
+							<div class="inline field {{if .Err_Auth}}error{{end}}">
+								<label for="mirror_username">{{.i18n.Tr "username"}}</label>
+								<input id="mirror_username" name="mirror_username" value="{{.Mirror.Username}}" {{if not .mirror_username}}data-need-clear="true"{{end}}>
+							</div>
+							<input class="fake" type="password">
+							<div class="inline field {{if .Err_Auth}}error{{end}}">
+								<label for="mirror_password">{{.i18n.Tr "password"}}</label>
+								<input id="mirror_password" name="mirror_password" type="password" value="">
```

Comment:
I think we should fill in the password here:

It looks confusing without it and gives the impression that it didn't save to show nothing in the password box after update. 

Also it looks like the code above will take the password from the mirror URL if a form is saved without a password -- so if you enter a username password, click update settings, then click update settings again it would silently leave you with an empty password.

Seems the only goal is to avoid people looking over a shoulder and seeing a password so I think just using the actual password is OK here.

Reply:
This code preserved the password

```		
		if form.MirrorUsername != "" {
			password := form.MirrorPassword
			if password == "" && oldURL != nil && u.Hostname() == oldURL.Hostname() {
				password, _ = oldURL.User.Password()
			}
			u.User = url.UserPassword(form.MirrorUsername, password)
		}
```

However, it's fine we can just round-trip the password.